server/test_route.py

```diff
@@ -38,5 +38,75 @@ def test_fetch_emotion(client):
 # TODO: sentiment testing
 # def test_fetch_sentiment(client):
 
-# TODO: summary testing
-# def test_fetch_summary(client):
+# Medium length input (400 words) - tests the basic API call works, and we get a response that 'summarises', i.e. shorter than the input.
+def test_fetch_summary_input_medium(client):
+    from src.model.data_model import SummaryLengthOption
+    from test.summary_inputs.medium_len import text
+
+    # Define the input data for testing
+    input_data = {
+        "text": text,
+        "summary_len_option": SummaryLengthOption.DEFAULT.value
+        }
+
+    # Test case for a successful POST request to /api/summary/process
+    response = client.post("/api/summary/process", json=input_data)
+
+    # Assert the response status code is 200 OK (or the desired status code)
+    assert response.status_code == 200
+
+    # Assert the response has a summary component
+    assert 'summary' in response.json()
+
+    # Assert the summary length is shorter than the input (can probably refine to a specific ratio, but hard based on tokens)
+    assert len(response.json()['summary']) < 0.8 * len(text)
```

Comment:
You can refer to the actual summary generation function - take the upper limit

![image](https://github.com/SIS-Team-24/2023-SIS-Team-24/assets/44883385/83e29cd3-40ad-4c1d-94b1-9e656e384be3)

Reply:
Apply this to the both "short" and "long" testing

Reply:
The current tests only use  `SummaryLengthOption.DEFAULT`, so have just added a check that the output tokens is within the bounds (input_tokens // 4 , input_tokens // 2)